Update Hailo library to v4.20.1

sairon opened the pull request:
## Proposed change
<!--
  Thank you!

  If you're introducing a new feature or significantly refactoring existing functionality, 
  we encourage you to start a discussion first. This helps ensure your idea aligns with 
  Frigate's development goals.

  Describe what this pull request does and how it will benefit users of Frigate.
  Please describe in detail any considerations, breaking changes, etc. that are 
  made in this pull request.
-->

The upcoming HAOS 15.0 will use Hailo driver v4.20.1. The current dev has the older patch release (v4.20.0) staged for future versions, but since new Frigate build will be needed for future compatibility, it makes sense to sync at the latest one.

Link: https://github.com/home-assistant/operating-system/pull/3922

## Type of change

- [x] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code
- [ ] Documentation Update

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 

## Checklist

<!--
  Put an `x` in the boxes that apply.
-->

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] The code has been formatted using Ruff (`ruff format frigate`)

sairon committed:
Update Hailo library to v4.20.1

The upcoming HAOS 15.0 will use Hailo driver v4.20.1. The current dev
has the older patch release (v4.20.0) staged for future versions, but
since new Frigate build will be needed for future compatibility, it
makes sense to sync at the latest one.

Link: https://github.com/home-assistant/operating-system/pull/3922

NickM-27 commented:
Thanks for letting us know, the first change will need to be done to https://github.com/frigate-nvr/hailort/tree/release_build

NickM-27 commented:
yes, discussion would be the right place

sairon commented:
@NickM-27 Okay, here it is:
- https://github.com/blakeblackshear/frigate/discussions/17096

sairon commented:
For the record, HA OS 15.0 was released today. As I tried to make it work with current stable Frigate, I discovered couple of more issues with the build of Hailo runtime binaries. I addressed the issues in my fork and summarized it in the discussion thread linked above.

scyto commented:
please consider linking in the stable versions docs to the old script for stable, took me a while to figure out my issues were the script kept installing the new version driver thats incompatible with stabele docker tags.....

not to mention the device didn't work inside a VM (i know it should, havent figured out why, have a question open on the hailo communuty, will update discussisons if i get a positive solution) everything worked fine when installed on my baremental host
